applications/Temp/temp.cpp

```diff
@@ -0,0 +1,36 @@
+
+#include "Base/MeshManipulator.h"
+#include "Base/CommandLineOptions.h"
+#include "Geometry/Mappings/RefinementMapsForCube.h"
+
+auto& meshName = Base::register_argument<std::string>(
+    'm', "meshName", "name of the Mesh file", true);
+
+int main(int argc, char* argv[]) {
+    Base::parse_options(argc, argv);
+
+    auto* configData = new Base::ConfigurationData(1, 0);
```

Comment:
Why not use unique_ptr?

Reply:
While a good idea, it would require significant refactoring of the kernel and other applications. So I would say that it would fall outside the scope of this pull request.

Sidenote, in my opinion the need for this configuration data should be removed completely. However, that can probably only be done by changing quite a bit of the architecture.

Reply:
I mean you can wrap the `new` call in a unique_ptr and use the `get()` method to pass the raw pointer along, than you do not have to call `delete` in the end

Reply:
Hmm, thinking of it. I'm wondering why it was even allocated via new instead of just put on the stack. Anyways, it should not even be in this pull request.